File: src/main/java/org/goafabric/personservice/adapter/CalleeServiceAdapter.java
package org.goafabric.personservice.adapter;

import org.eclipse.microprofile.faulttolerance.CircuitBreaker;
import org.eclipse.microprofile.faulttolerance.Timeout;
import org.eclipse.microprofile.rest.client.annotation.RegisterClientHeaders;
import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;

import jakarta.ws.rs.GET;
import jakarta.ws.rs.Path;
import jakarta.ws.rs.PathParam;
import jakarta.ws.rs.QueryParam;

@Path("/callees")
@RegisterRestClient
@Timeout
@CircuitBreaker
@RegisterClientHeaders(AdapterConfiguration.class)
public interface CalleeServiceAdapter {
    @GET
    @Path("sayMyName")
    Callee sayMyName(@QueryParam("name") String name);

    @GET
    @Path("sayMyOtherName/{name}")
    Callee sayMyOtherName(@PathParam("name") String name);

    @GET
    @Path("setSleepTime")
    Callee setSleepTime(@QueryParam("sleepTime") Long sleepTime);
}

File: src/main/kotlin/org/goafabric/personservice/adapter/CalleeServiceAdapter.kt
package org.goafabric.personservice.adapter

import jakarta.enterprise.context.ApplicationScoped
import jakarta.ws.rs.GET
import jakarta.ws.rs.Path
import jakarta.ws.rs.PathParam
import jakarta.ws.rs.QueryParam
import org.eclipse.microprofile.faulttolerance.CircuitBreaker
import org.eclipse.microprofile.faulttolerance.Timeout
import org.eclipse.microprofile.rest.client.annotation.RegisterClientHeaders
import org.eclipse.microprofile.rest.client.inject.RegisterRestClient

@Path("/callees")
@RegisterRestClient
@Timeout
@CircuitBreaker
@RegisterClientHeaders(AdapterConfiguration::class)
@ApplicationScoped
interface CalleeServiceAdapter {
    @GET
    @Path("sayMyName")
    fun sayMyName(@QueryParam("name") name: String): Callee

    @GET
    @Path("sayMyOtherName/{name}")
    fun sayMyOtherName(@PathParam("name") name: String): Callee

    @GET
    @Path("setSleepTime")
    fun setSleepTime(@QueryParam("sleepTime") sleepTime: Long): Callee
}
